source/gui/addonStoreGui/views.py

```diff
@@ -0,0 +1,609 @@
+
+	# noinspection PyMethodMayBeStatic
+	def OnItemActivated(self, evt: wx.ListEvent):
+		activatedIndex = evt.GetIndex()
+		log.debug(f"item activated: {activatedIndex}")
+
+	def OnItemDeselected(self, evt: wx.ListEvent):
+		log.debug(f"item deselected")
+		self._addonsListVM.setSelection(None)
+
+	def OnGetItemText(self, itemIndex: int, colIndex: int):
+		dataItem = self._addonsListVM.getAddonAttrText(
+			itemIndex,
+			AddonListVM.presentedAttributes[colIndex]
+		)
+		return str(dataItem)
+
+	def OnColClick(self, evt: wx.ListEvent):
+		colIndex = evt.GetColumn()
+		log.debug(f"col clicked: {colIndex}")
+		self._addonsListVM.setSortField(AddonListVM.presentedAttributes[colIndex])
+
+	def _doRefresh(self):
+		with guiHelper.autoThaw(self):
+			newCount = self._addonsListVM.getCount()
+			self.SetItemCount(newCount)
+			self._refreshSelection()
+
+	def _refreshSelection(self):
+		selected = self.GetFirstSelected()
+		newSelectedIndex = self._addonsListVM.getSelectedIndex()
+		log.debug(f"_refreshSelection {newSelectedIndex}")
+		if newSelectedIndex is not None:
+			self.Select(newSelectedIndex)
+			self.Focus(newSelectedIndex)
+			# wx.ListCtrl doesn't send a selection event if the index hasn't changed,
+			# however, the item at that index may have changed as a result of filtering.
+			# To ensure parent dialogs are notified, explicitly send an event.
+			if selected == newSelectedIndex:
+				evt = wx.ListEvent(wx.wxEVT_LIST_ITEM_SELECTED, self.GetId())
+				evt.SetIndex(newSelectedIndex)
+				evt.SetClientObject(self._addonsListVM.getSelection())
+				self.GetEventHandler().ProcessEvent(evt)
+		elif newSelectedIndex is None:
+			# wx.ListCtrl doesn't send a deselection event when the list is emptied.
+			# To ensure parent dialogs are notified, explicitly send an event.
+			self.Select(selected, on=0)
+			evt = wx.ListEvent(wx.wxEVT_LIST_ITEM_DESELECTED, self.GetId())
+			evt.SetIndex(-1)
+			evt.SetClientObject(None)
+			self.GetEventHandler().ProcessEvent(evt)
+
+
+def _insertLinkForLabelWithTomViaComIDispatch(
+		ctrl: wx.TextCtrl,
+		labelText: str,
+		linkText: str,
+		url: str,
+) -> None:
+	""" Use Text Object model via COM IDispatch to insert a link at the end of the TextCtrl content.
+	The underlying control must be a win32 RichEdit control, such as RichEdit50.
+	Used because wx.TextCtrl does not expose adding URLs.
+
+	@param ctrl: The control to work with.
+	@param labelText: The label text to find and insert a link after.
+	@param linkText: The text to append.
+	@param url: The URL to link to.
+	"""
+	initialEditableState = ctrl.IsEditable()
+	ctrl.SetEditable(True)
+	import oleacc
+	import winUser
+	import comtypes.automation
+	import comtypes.client
+	import comInterfaces.tom
+	hwnd = ctrl.GetHandle()
+	obj = oleacc.AccessibleObjectFromWindow(
+		hwnd,
+		winUser.OBJID_NATIVEOM,
+		interface=comtypes.automation.IDispatch
+	)
+	# No definitions available (in NVDA) for ITextRange2 interface, I.E. they are not in the msftedit type

```

Comment:
Have you looked into this? I recall I did at some point, but no longer recall how and why. I guess it is preferred to use the proper interface directly.

Reply:
No, I think I had a discussion with Mick about it at the time, but it was really beyond the scope of this work.